torch/_inductor/codegen/wrapper.py

```diff
@@ -156,10 +156,15 @@ def codegen(self, code: IndentedBuffer, device_cm_stack: contextlib.ExitStack):
                 # associated with a device, so we never expect the device to change.
                 assert self.first_time
                 # CUDAStreamGuard sets the stream and the device.
-                code.writeline(
-                    f"at::cuda::CUDAStreamGuard stream_guard("
-                    f"at::cuda::getStreamFromExternal(stream, {self.device_idx}));"
-                )
+                if config.aot_inductor.abi_compatible:
+                    code.writeline(
+                        f"AOTI_TORCH_ERROR_CHECK(aoti_torch_set_current_cuda_stream(stream, {self.device_idx}));"
```

Comment:
This doesn't seem like an exact equivalence; in particular, the old code would reset the stream back to its original setting at the end of execution (or on early exit.) Another RAII wrapper here would make it easy to replicate the correct semantics!